awx/main/models/inventory.py

```diff
@@ -1801,3 +1816,891 @@ class Meta:
+        if credential is None:
+            return injected_env
+        if self.base_injector == 'managed':
+            from awx.main.models.credential import injectors as builtin_injectors
+            cred_kind = inventory_update.source.replace('ec2', 'aws')
+            if cred_kind in dir(builtin_injectors):
+                getattr(builtin_injectors, cred_kind)(credential, injected_env, private_data_dir)
+                if safe:
+                    from awx.main.models.credential import build_safe_env
+                    return build_safe_env(injected_env)
+        elif self.base_injector == 'template':
+            injected_env['INVENTORY_UPDATE_ID'] = str(inventory_update.pk)  # so injector knows this is inventory
+            safe_env = injected_env.copy()
+            args = []
+            safe_args = []
+            credential.credential_type.inject_credential(
+                credential, injected_env, safe_env, args, safe_args, private_data_dir
+            )
+            if safe:
+                return safe_env
+        return injected_env
+
+    def get_plugin_env(self, inventory_update, private_data_dir, private_data_files, safe=False):
+        return self._get_shared_env(inventory_update, private_data_dir, private_data_files, safe)
+
+    def get_script_env(self, inventory_update, private_data_dir, private_data_files, safe=False):
+        injected_env = self._get_shared_env(inventory_update, private_data_dir, private_data_files, safe)
+
+        # Put in env var reference to private ini data files, if relevant
+        if self.ini_env_reference:
+            credential = inventory_update.get_cloud_credential()
+            cred_data = private_data_files.get('credentials', '')
+            injected_env[self.ini_env_reference] = cred_data[credential]
+
+        return injected_env
+
+    def build_private_data(self, inventory_update, private_data_dir):
+        if self.should_use_plugin():
+            return self.build_plugin_private_data(inventory_update, private_data_dir)
+        else:
+            return self.build_script_private_data(inventory_update, private_data_dir)
+
+    def build_script_private_data(self, inventory_update, private_data_dir):
+        return None
+
+    def build_plugin_private_data(self, inventory_update, private_data_dir):
+        return None
+
+    @staticmethod
+    def dump_cp(cp, credential):
+        """Dump config parser data and return it as a string.
+        Helper method intended for use by build_script_private_data
+        """
+        if cp.sections():
+            f = StringIO()
+            cp.write(f)
+            private_data = private_data = {'credentials': {}}
+            private_data['credentials'][credential] = f.getvalue()
+            return private_data
+        else:
+            return None
+
+
+class azure_rm(PluginFileInjector):
+    plugin_name = 'azure_rm'
+    # FIXME: put in correct version when Ansible core fixes are in
```

Comment:
This comment still accurate given https://github.com/ansible/ansible/pull/53046 ?

Reply:
Good comment. I will push a change that updates all these to the final status, now that they are resolved. There are different reasons that each needs to be 2.8, so I want those reasons in a brief code comment, in case some cowboy thinks they can back it off without consequence.